examples/inverse-kinematics.cpp

```diff
@@ -39,9 +41,12 @@ int main(int /* argc */, char ** /* argv */)
       break;
     }
     pinocchio::computeJointJacobian(model,data,q,JOINT_ID,J);
-    const Eigen::VectorXd v     = - svd.compute(J.topRows<3>()).solve(err);
+    pinocchio::SE3::Matrix6 Jlog;
+    pinocchio::Jlog6(dMi, Jlog);
+    v = - svd.compute(Jlog * J).solve(err);
```

Comment:
You should change the way you've defined the error to not use the Jlog in your expression.
Indeed, using Jlog may trouble people that are only familiar with standard IK.

So, the error should be:
```
     const pinocchio::SE3 dMi = data.oMi[JOINT_ID].actInv(oMdes); 
```
giving rise:
```
v = + svd.compute(J).solve(err);
```
If I'm not making mistakes ;)

Reply:
@jcarpent, then how would you convert `dMi` into a vector? The purpose of `log` here is performing this operation.
Indeed, I thought that using log was a plus. I, in the first place, was not familiar with the use of log in the inverse kinematics example, and I had to find out myself how it all worked when I had to implement an inverse kinematics of my own; hence the idea of putting it in the example. I would have appreciated the existence of this example as it is the way I am presenting it, and so I think other people will.
Of course we are not obliged to pass through the log, there are other ways. Examples can be found in Sec. 3.7.3 of Oriolo's book "Robotics Modelling, Planning and Control" ("Orientation error"), subsections "Angle axis" and "Unit Quaternion" (I can send you a link to the book if you do not have it and you wish to take a look). But it would involve lengthier computations and possibly defining dedicated functions for the error. Instead, here, I am showing how everything can be done in Pinocchio using just a few lines, employing only Pinocchio functions. I think this makes the example both more compact and more significant. I am explaining the use of the log in the text (that example was already commented for position-only error, I corrected the relevant parts).

Reply:
In you example, as you can notice, and keep the error as:
```
err = pinocchio::log6(dMi).toVector()
```

Reply:
You just change the way the error is represented. On your side, your expressed the error with respect to the target frame. On my side, it expressed in the local frame of the end-effector. Two opposite view of the same thing ;)

Reply:
Oh! Ok, I understand, I will do it immediately

Reply:
Actually, are you sure? I think it is easier to understand my way. In 3D, it was:
```
err = R.transpose()*(x-xdes);
v = - svd.compute(J.topRows<3>()).solve(err);	
```
which I find better than
```
err = R.transpose()*(xdes-x);
v = svd.compute(J.topRows<3>()).solve(err);	
```

Reply:
Relying on Jlog may not be easy to get for most people.

Reply:
I confirm that without Jlog it does not work, even changing the expression of the error. This does not surprise me, because `log(M^-1) = - log(M)`, so using `data.oMi[JOINT_ID].actInv(oMdes)` or `oMdes.actInv(data.oMi[JOINT_ID])` should not change anything.
I think we need to stuck to `Jlog`. So I would say this PR is ready

Reply:
You don't have any model with exponential decay on your system. Inverse kinematics is different from inverse geometry !!!

Reply:
This is the way I have always done and it works. I could sketch the equations, but I would rather not do it, I do not have time for that and I do not think it is needed.
With your proposed changes, instead, the example does not work anymore. And I do not see how it could be different, because if you incorporate log in the expression of the error, you need the derivative on the log.
I could switch to a Jacobian-transpose system, but I do not think it would change anything and I do not see the point of it.
I remind you that this is just an example of how to use the code with no intention of being 100% rigorous and functional.
For me, the question is settled. If you wish to introduce changes, you are welcome to do it yourself

Reply:
@jcarpent, what about this?
I do not have more time to spend on this. If you do not like the changes, I can just remove the two commits involving the 6D inverse kinematics, and the example will stay in 3D. Then you can merge the rest.